src/nmr_to_structure/prepare_input/prepare_nmr_rxn_input.py

```diff
@@ -1,92 +1,163 @@
+    combined_df = pd.concat(return_list)
 
-        nmr_string = " " + hnmr_string
+    logger.info("Printing Tanimoto statistics:")
+    for key in pd.unique(combined_df.tanimoto_bin):
+        logger.info(
+            "Tanimoto bin {}: {:.3f}%".format(
+                key, (combined_df.tanimoto_bin == key).sum() / len(combined_df) * 100
+            )
+        )
 
-    elif mode == "cnmr":
-        cnmr_string = build_cnmr_string(
-            cnmr,
-            header=False,
-            token_space=token_space,
-        )[0]
-        nmr_string = " " + cnmr_string
-    
-    return nmr_string
+    return combined_df
 
 
 def make_nmr_rxn_set(
-    nmr_set: pd.DataFrame,
+    nmr_df: pd.DataFrame,
     rxn_set: pd.DataFrame,
+    n_samples: int,
     mode: str,
     hnmr_mode: str = "range",
-    token_space: str ="shared",
-    non_matching: bool = True
-
+    token_space: str = "shared",
+    non_matching: bool = True,
 ) -> pd.DataFrame:
     rxns = list(rxn_set.index)
-    mols = set(nmr_set.index)
+    random.shuffle(rxns)
+
+    mols = set(nmr_df.index)
     mols_sample = tuple(mols)
 
-    src_tgt_pairs = list()
+    src_tgt_pairs: List[dict] = list()
 
-    for rxn in tqdm(rxns):
-        rxn = rxn.replace(">", ".").replace("..", "")
+    idx = 0
+    pbar = tqdm(total=n_samples)
+    while len(src_tgt_pairs) < n_samples:
+        rxn = rxns[idx].replace(">", ".").replace("..", "")
         components = rxn.split(".")
         components = list(filter(None, components))
 
-        components_tokenized = tokenize_smiles(".".join(components))
+        allowed_mask = [evaluate_molecule(smiles) for smiles in components]
+        allowed_components = [
+            components[i]
+            for i in range(len(allowed_mask))
+            if allowed_mask[i] and len(components[i]) > 5
+        ]
 
-        for component in components:
+        components_tokenized = tokenize_smiles(".".join(allowed_components))
+
+        for component in allowed_components:
             if component not in mols or len(component) < 5:
                 continue
-            
+
             # Make matching set
             nmr_input = components_tokenized
-            nmr = make_nmr(mode, component, 
-                           hnmr=nmr_set.loc[component]["1H_NMR_sim"]["peaks"], 
-                           cnmr=nmr_set.loc[component]["13C_NMR_sim"], 
-                           hnmr_mode=hnmr_mode,
-                           token_space=token_space)
-            nmr_input = components_tokenized + ' ' + nmr
+            nmr = make_nmr(
+                mode,
+                component,
+                hnmr=nmr_df.loc[component]["1H_NMR_sim"]["peaks"],
+                cnmr=nmr_df.loc[component]["13C_NMR_sim"],
+                hnmr_mode=hnmr_mode,
+                token_space=token_space,
+            )
+            nmr_input = components_tokenized + nmr
+
+            if nmr_input.count(" ") + 1 > DEFAULT_MAX_SEQ_LEN:
+                continue
```

Comment:
Add a small comment to say (if I understand correctly) that too long sequences are ignored?

Depending on how often this happens, you may want to consider adding a `logger.info` or so